velox/functions/sparksql/tests/MapTest.cpp

```diff
@@ -105,5 +106,122 @@ TEST_F(MapTest, ErrorCases) {
       {inputVector1, inputVector2, inputVector11, inputVector22},
       true);
 }
+
+TEST_F(MapTest, complexTypes) {
+  auto rowKey = makeRowVector({
+      makeFlatVector<int64_t>(1, [](auto row) { return row; }),
+  });
+  auto rowValue = makeRowVector({
+      makeFlatVector<int64_t>(1, [](auto row) { return row; }),
+  });
+  auto mapRowResult = makeMapVector(
+      {
+          0,
+      },
+      rowKey,
+      rowValue);
+  mapSimple("map(c0, c1)", {rowKey, rowValue}, false, mapRowResult);
+
+  auto arrayKey = makeArrayVectorFromJson<int64_t>({"[1, 2, 3]"});
+  auto arrayValue = makeArrayVectorFromJson<int64_t>({"[1, 2, 3]"});
+  auto arrayMapResult = makeMapVector(
+      {
+          0,
+      },
+      arrayKey,
+      arrayValue);
+  mapSimple("map(c0, c1)", {arrayKey, arrayValue}, false, arrayMapResult);
+
+  auto mapKey = makeMapVector(
+      {
+          0,
+      },
+      rowKey,
+      rowValue);
+  auto mapValue = makeMapVector(
+      {
+          0,
+      },
+      rowKey,
+      rowValue);
+  auto mapResult = makeMapVector(
+      {
+          0,
+      },
+      mapKey,
+      mapValue);
+  mapSimple("map(c0, c1)", {mapKey, mapValue}, false, mapResult);
+
+  auto nestedMapKey = makeMapVector(
+      {
+          0,
+      },
+      mapKey,
+      rowValue);
+  auto nestedMapValue = makeMapVector(
+      {
+          0,
+      },
+      arrayKey,
+      mapValue);
+  auto nestedMapResult = makeMapVector(
+      {
+          0,
+      },
+      nestedMapKey,
+      nestedMapValue);
+  mapSimple(
+      "map(c0, c1)", {nestedMapKey, nestedMapValue}, false, nestedMapResult);
+
+  auto nullArrayValue = makeNullableArrayVector<int64_t>({
+      std::nullopt,
+  });
+  auto nullValueMapResult = makeMapVector(
+      {
+          0,
+      },
+      arrayKey,
+      nullArrayValue);
+  mapSimple(
+      "map(c0, c1)", {arrayKey, nullArrayValue}, false, nullValueMapResult);
+
+  auto mixedArrayKey1 = makeArrayVector<int64_t>({{1, 2, 3}});
+
+  auto mixedRowValue1 = makeRowVector({
+      makeFlatVector<int32_t>(1, [](auto row) { return row; }),
```

Comment:
This code repeats a lot. Consider extracting a helper function or better test with more than 1 element vector.

```
makeFlatVector<int32_t>({1, 2, 3})
```

Reply:
Updated, thanks.